stpsf/utils.py

```diff
@@ -216,30 +220,32 @@ def auto_download_stpsf_data():
 def get_stpsf_data_path(data_version_min=None, return_version=False):
     """Get the STPSF data path
 
-    Simply checking an environment variable is not always enough, since
-    for packaging this code as a Mac .app bundle, environment variables are
-    not available since .apps run outside the Terminal or X11 environments.
-
-    Therefore, check first the environment variable STPSF_PATH, and secondly
+    Check first the environment variable STPSF_PATH, and secondly
     check the configuration file in the user's home directory.
 
     If data_version_min is supplied (as a 3-tuple of integers), it will be
     compared with the version number from version.txt in the STPSF data
     package.
     """
-    import os
-    from pathlib import Path
 
     path_from_config = conf.STPSF_PATH  # read from astropy configuration
     if path_from_config == 'from_environment_variable':
         path = os.getenv('STPSF_PATH')
         if path is None:
-            message = (
-                'Environment variable $STPSF_PATH is not set!\n'
-                f'{MISSING_STPSF_DATA_MESSAGE} searching default location..s'
-            )
-            warnings.warn(message)
-            path = auto_download_stpsf_data()
+
+            # Check if the data were already downloaded to the default path
+            default_path = get_default_data_path()
```

Comment:
This change isn't necessary as this logic already exists within `auto_download_stpsf_data `(which also sets the STPSF_PATH which is necessary)

`auto_download_stpsf_data()` will make a new default directory (only if one doesn't exist), and download the data (only if the directory is empty).

Reply:
My point here was intentionally refactoring to allow _checking for a prior successful call_ to auto_download_stpsf_data, prior to displaying the big giant warning message about trying to download the data. If the data has already been downloaded successfully, it’s incorrect to display a warning that says “stpsf will try to download the data”. 

The current code will display the warning text every time the package is imported in a new Python session, even if the data is already indeed present and fine in the default directory. That’s excess warnings for no benefit to the user.

Reply:
I see your concern (deleted my previous reply because I misunderstood what download message you were referring to)

I think its important that the `Environment variable $STPSF_PATH is not set!` warning is still displayed.  

May just make more sense to move the `MISSING_STPSF_DATA_MESSAGE` inside `auto_download_stpsf_data` inside the `if not any(default_path.iterdir()):` codeblock.